src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/ApplicationTest.kt

```diff
@@ -1872,295 +1874,426 @@ class ApplicationTest : IntegrationTestBase() {
+        .header("Authorization", "Bearer $jwt")
+        .bodyValue(
+          SubmitApprovedPremisesApplication(
+            noticeType = Cas1ApplicationTimelinessCategory.standard,
+            apType = ApType.normal,
+            translatedDocument = {},
+            isWomensApplication = false,
+            targetLocation = "SW1A 1AA",
+            releaseType = ReleaseTypeOption.licence,
+            sentenceType = SentenceTypeOption.nonStatutory,
+            type = "CAS1",
+            applicantUserDetails = Cas1ApplicationUserDetails(
+              "applicantName",
+              "applicantEmail",
+              "applicationTelephone",
             ),
-          )
+            caseManagerIsNotApplicant = false,
+            caseManagerUserDetails = Cas1ApplicationUserDetails(
+              "cmName",
+              "cmEmail",
+              "cmTelephone",
+            ),
+            reasonForShortNotice = "reasonForShort",
+            reasonForShortNoticeOther = "reasonForShortOther",
+          ),
+        )
+        .exchange()
+        .expectStatus()
+        .isOk
+
+      val persistedApplication = approvedPremisesApplicationRepository.findByIdOrNull(applicationId)!!
+
+      assertThat(persistedApplication.noticeType).isEqualTo(Cas1ApplicationTimelinessCategory.standard)
+      assertThat(persistedApplication.isWomensApplication).isFalse()
+      assertThat(persistedApplication.isPipeApplication).isFalse
+      assertThat(persistedApplication.targetLocation).isEqualTo("SW1A 1AA")
+      assertThat(persistedApplication.sentenceType).isEqualTo(SentenceTypeOption.nonStatutory.toString())
+      assertThat(persistedApplication.apArea?.id).isEqualTo(submittingUser.apArea!!.id)
+
+      val createdAssessment =
+        approvedPremisesAssessmentRepository.findAll().first { it.application.id == applicationId }
+      assertThat(createdAssessment.allocatedToUser).isNull()
+      assertThat(createdAssessment.createdFromAppeal).isFalse()
+
+      val persistedDomainEvents = domainEventRepository.findAll().filter { it.applicationId == applicationId }
+      assertThat(persistedDomainEvents).hasSize(1)
+      val persistedApplicationSubmittedEvent = persistedDomainEvents.firstOrNull { it.type == DomainEventType.APPROVED_PREMISES_APPLICATION_SUBMITTED }
+
+      assertThat(persistedApplicationSubmittedEvent).isNotNull
+      assertThat(persistedApplicationSubmittedEvent!!.crn).isEqualTo(offenderDetails.otherIds.crn)
+
+      val emittedMessage = domainEventAsserter.blockForEmittedDomainEvent(DomainEventType.APPROVED_PREMISES_APPLICATION_SUBMITTED)
+
+      val emittedMessageDescription = "An application has been submitted for an Approved Premises placement"
+      assertThat(emittedMessage.description).isEqualTo(emittedMessageDescription)
+      assertThat(emittedMessage.detailUrl).matches("http://api/events/application-submitted/[0-9a-fA-F]{8}\\b-[0-9a-fA-F]{4}\\b-[0-9a-fA-F]{4}\\b-[0-9a-fA-F]{4}\\b-[0-9a-fA-F]{12}")
```

Comment:
Do we need regex here?  Is this not the application id?

Reply:
Oh yes, good spot. A lot of existing tests do this when the IDs are actually known, i've no idea why! i'll change it to use the actual ID

Reply:
oops, i was wrong here. It's actually the domain event id, but we know that so may as well use it instead of a regex!